src/Cafe/HW/Latte/Renderer/Vulkan/VulkanPipelineCompiler.cpp

```diff
@@ -988,7 +988,8 @@ bool PipelineCompiler::Compile(bool forceCompile, bool isRenderThread, bool show
 	pipelineInfo.pDynamicState = &dynamicState;
 	pipelineInfo.pRasterizationState = &rasterizer;
 	pipelineInfo.pMultisampleState = &multisampling;
-	pipelineInfo.pColorBlendState = &colorBlending;
+	if (!_IsVkIntegerFormat(m_renderPassObj->GetColorFormat(0)))
+		pipelineInfo.pColorBlendState = &colorBlending;
```

Comment:
This should iterate each color attachment and when an integer format is used then disable blending only for that specific attachment. Having one color attachment in integer format does not prevent blending for all the non-integer attachments.

Reply:
I see. I don't want to chalk this down as a bug in MoltenVk yet because it seems to be too big of an oversight to me. I checked [their code](https://github.com/KhronosGroup/MoltenVK/blob/3b9e335fe8fc8a72cf6099acbd54128889445c0a/MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm#L1689) and I couldn't really see any logic that would enable blending if we didn't specifically ask for it. But more digging is required to understand this fully.

In the meantime an acceptable temporary workaround would be to disable all blending if ANY of the color attachments are integer format instead of only checking index 0. I also suggest making the workaround apply to MacOS/MoltenVk only.